File: src/Add_UserInfo.cc
#include <TFile.h>
#include <TTree.h>
#include <TParameter.h>

void Add_UserInfo(char filename[]){
    // TFile *file = TFile::Open("../data/residual_all.root", "UPDATE");
    TFile *file = TFile::Open(filename, "UPDATE");

    for (int i = 0; i <6;i++) {
        TTree *data = (TTree*)file->Get(Form("nHit%02d/data",i));
        TTree *bkg = (TTree*)file->Get(Form("nHit%02d/bkg",i));

        TParameter<int> *obj1 = new TParameter("Nside",1024);
        TParameter<int> *obj2 = new TParameter("Scheme",0);
        
        data->GetUserInfo()->Add(obj1);
        data->GetUserInfo()->Add(obj2);
        bkg->GetUserInfo()->Add(obj1);
        bkg->GetUserInfo()->Add(obj2);
    }
    file->Write();
    file->Close();
}

File: src/Add_UserInfo.py
import ROOT

file = ROOT.TFile.Open("./residual_all.root", "Updates")
for bin in range(6):
    
    tdata=file.Get("nHit%02d"%int(bin)).data
    tbkg=file.Get("nHit%02d"%int(bin)).bkg

    obj1=ROOT.TParameter(int)("Nside",1024)
    obj2=ROOT.TParameter(int)("Scheme",0)

    tdata.GetUserInfo().Add(obj1)
    tdata.GetUserInfo().Add(obj2)
    tbkg.GetUserInfo().Add(obj1)
    tbkg.GetUserInfo().Add(obj2)

file.Write("")
file.Close()